File: C++/SegundoParcial/Practico_5/Punto10.cpp
/* Escribir un programa que le permita al usuario ingresar la diagonal principal de una matriz
de 5x5 y que los otros lugares se completen con una X.*/

#include <iostream>

using namespace std;

int main(){

    int matriz[5][5]; 
    
    //Cargamos la MATRIZ con un condicional 
    for (int i = 0; i < 5; i++){
        for (int j = 0; j < 5; j++){
            if (i == j){
                cout << "\nIngrese numero para la matriz diagonal: ";
                cin >> matriz[i][j];
            }
        }
    }


    //Mostramos la MATRIZ
    for (int i = 0; i < 5; i++){
        for (int j = 0; j < 5; j++){
            if (i == j)
            {
                cout << matriz[i][j] << " ";
            }
            else
            {
                cout << "X "; 
            }
        }
        cout << endl;
    }

    return 0;

}

File: Python3/PrimerParcial/Practico 1/Punto10.py
"""
Escriba un programa que permita al usuario ingresar las medidas de 2 lados de un
rectángulo, y que luego mediante la impresión repetida de un caracter (ej: *) lo dibuje
en la pantalla. Para este ejercicio tomaremos un máximo de 40 para el lado más
largo, con el fin de evitar problemas de visualización en la consola. Verificar en los
datos de entrada que se cumpla este requisito.
"""

ancho= int(input("\nIngrese medida del primer lado: "))
alto = int(input("Ingrese medida del segundo lado: "))

while (alto > 40) or (ancho > 40):
    print("\n[!]Ingresa un numero menor a 40")
    alto= int(input("\nIngrese medida del primer lado: "))
    ancho = int(input("Ingrese medida del segundo lado: "))

#Dibujar el rectangulo
for i in range(alto):
    print("*" * ancho)